tests/benchmark.py

```diff
@@ -3,53 +3,125 @@
+        with open(filename, 'r') as rxn_file:
+            for line in rxn_file:
+                solvent = None if len(line.split()) < 3 else line.split()[2]
+                rxn = ade.Reaction(smiles=line.split()[1],
+                                   name=line.split()[0],
+                                   solvent_name=solvent)
+                reactions.append(rxn)
 
-cope = Reaction(Reactant(name='diene1', smiles='C=C[C@H](C)[C@@H](C)C=C'),
-                Product(name='diene2', smiles='C/C=C/CC/C=C/C'),
-                name='cope_rearrangement')
+        return None
 
-da = Reaction(Reactant(name='butadiene', smiles='C=CC=C'),
-              Reactant(name='ethene', smiles='C=C'),
-              Product(name='cyclohexene', smiles='C1C=CCCC1'),
-              name='diels_alder')
+    def add_xyz_rxns_from_file(filename):
+        """Add reactions from a file with lines in the format:
+         name  XX.YY>>ZZ"""
 
-h_shift = Reaction(Reactant(name='radical1', smiles='CC[C]([H])[H]'),
-                   Product(name='radical2', smiles='C[C]([H])C'),
-                   name='h_shift')
+        with open(filename, 'r') as rxn_file:
+            for line in rxn_file:
+                name, rxn_str = line.split()
+                reac_names, prod_names = rxn_str.split('>>')
+                reacs = [ade.Reactant(os.path.join(data_path, f'{name}.xyz'))
+                         for name in reac_names.split('.')]
+                prods = [ade.Product(os.path.join(data_path, f'{name}.xyz'))
+                         for name in prod_names.split('.')]
 
-int1_xyz = os.path.join(data_path, 'INT1.xyz')
-int2_xyz = os.path.join(data_path, 'INT2.xyz')
+                rxn = ade.Reaction(*reacs, *prods, name=name)
+                reactions.append(rxn)
 
-mig_insert = Reaction(Reactant(name='INT1', atoms=xyz_file_to_atoms(int1_xyz)),
-                      Product(name='INT2', atoms=xyz_file_to_atoms(int2_xyz)),
-                      name='h_insert')
+        return None
 
-print(f'Name            v_imag / cm-1       Time / min         Success')
-for reaction in [sn2, cope, da, h_shift, mig_insert]:
+    reactions = []
+    args = get_args()
 
-    start_time = time()
-    reaction.locate_transition_state()
-    freq = reaction.ts.imaginary_frequencies[0]
+    if args.smallorganic or args.all:
+        add_smiles_rxns_from_file(os.path.join(data_path, 'ADE_SO.txt'))
 
-    print(f'{reaction.name:.15}'
-          f'{freq:15.1f}'
-          f'{(time()- start_time)/60:15.1f}'
-          f'{"✓" if freq < -100 else "✗":.15s}')
+    if args.smallmetal or args.all:
+        add_xyz_rxns_from_file(os.path.join(data_path, 'ADE_SM.txt'))
+
+    if len(reactions) == 0:
+        raise StopIteration('Had no reactions to enumerate. Call this script ' 
+                            'with e.g. --smallorganic. Run '
+                            '*python benchmark.py --help* for all options')
+    return reactions
+
+
+if __name__ == '__main__':
```

Comment:
should this have some way of checking if the calcs are already run?

Reply:
I don't think there's any need(?)